runtime/dart_init.cc

```diff
@@ -188,11 +193,13 @@ Dart_Isolate ServiceIsolateCreateCallback(const char* script_uri,
   // No VM-service in release mode.
   return nullptr;
 #else   // FLUTTER_RUNTIME_MODE
-  tonic::DartState* dart_state = new tonic::DartState();
-  Dart_Isolate isolate = Dart_CreateIsolate(
-      script_uri, "main", g_default_isolate_snapshot_data,
-      g_default_isolate_snapshot_instructions, nullptr, dart_state, error);
+  blink::UIDartState* dart_state = new blink::UIDartState(nullptr, nullptr);
+  Dart_Isolate isolate =
+      Dart_CreateIsolate(script_uri, "main", g_default_isolate_snapshot_data,
+                         g_default_isolate_snapshot_instructions, nullptr,
+                         static_cast<tonic::DartState*>(dart_state), error);
```

Comment:
Is static cast actually needed here?

Reply:
That's how it's done in `dart_controller.cc` so that's what I did here. I thought it was important to use static_cast when casting between sub and super classes.